irods/test/PRC_issue_362.bats

```diff
@@ -12,25 +13,43 @@ setup() {
   [ $BATS_TEST_NUMBER = 1 ] && echo "---" >/tmp/PRC_test_issue_362
   local name=${BATS_TEST_DESCRIPTION##*_}
   CHR="${chars[$name]}"
-}
 
-TEST_THE_TEST=""
+  ## Arrange for secrets file to be generated internally by the Python client
+  cat >~/.python_irodsclient <<-EOF
+	legacy_auth.pam.store_password_to_environment True
+	legacy_auth.pam.password_for_auto_renew 'my${CHR}pass'
+	legacy_auth.pam.time_to_live_in_hours 1
+	EOF
+
+  iinit_as_rods
+
+  if [ ! -e /tmp/rodsuser_alissa_created ]; then
+      iadmin mkuser alissa rodsuser
+  fi
+  touch /tmp/rodsuser_alissa_created
+
+  _begin_pam_environment_and_password "" alissa
+  rm -f ~/.irods/.irodsA
+
+  cat >~/test_get_home_coll.py <<-EOF
+	import irods.test.helpers as h
+	ses = h.make_session()
+	home_coll = h.home_collection(ses)
+	exit(0 if ses.collections.get(home_coll).path == home_coll
+	       and ses.pool.account._original_authentication_scheme.lower().startswith('pam')
+	     else 1)
+	EOF
+}
 
 prc_test()
 {
   local USER="alissa"
-  local PASSWORD=$(tr "." "$CHR" <<<"my.pass")
-  echo "$USER:$PASSWORD" | sudo chpasswd
-  if [ "$TEST_THE_TEST" = 1 ]; then
-    echo -n `date`: "" >&2
-    { su - "$USER" -c "id" <<<"$PASSWORD" 2>/dev/null | grep $USER ; } >&2
-  else
-    sudo su - irods -c "env PYTHON_IRODSCLIENT_TEST_PAM_PW_OVERRIDE='$PASSWORD' python -m unittest \
-                        irods.test.login_auth_test.TestLogins.test_escaped_pam_password_chars__362"
-  fi
-} 2>> /tmp/PRC_test_issue_362
+  local PASSWORD="my${CHR}pass"
+  sudo chpasswd <<<"$USER:$PASSWORD"
+  env PYTHON_IRODSCLIENT_CONFIGURATION_PATH='' python ~/test_get_home_coll.py
+}
 
-@test "test_with_atsymbol"	{ prc_test; }
-@test "test_with_semicolon"	{ prc_test; }
-@test "test_with_equals"	{ prc_test; }
-@test "test_with_ampersand"	{ prc_test; }
+@test "test_with_atsymbol" { prc_test; }
+@test "test_with_semicolon" { prc_test; }
+@test "test_with_equals" { prc_test; }
+@test "test_with_ampersand" { prc_test; }
```

Comment:
Are the tests actually different? If yes, how/where is `prc_test` being changed?

Reply:
Its a hack actually where we change a global variable based on the name of the test, $BATS_TEST_NAME.

Yeah, ick.  I can change that easily

Reply:
Done